File: examples/ruby/pagination/sandbox.rb
# frozen_string_literal: true

require 'northrays'

northrays = Northrays::Northrays.new

northrays.list(Northrays::ListSandboxesQuery.new(
               limit: 10,
               labels: { 'env' => 'dev' },
               states: [Northrays::SandboxState::STARTED],
               sort: Northrays::SandboxListSortField::CREATED_AT,
               order: Northrays::SandboxListSortDirection::DESC
             )).each do |sandbox|
  puts sandbox.id
end

File: examples/python/pagination/sandbox.py
from northrays import Northrays, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


def main():
    northrays = Northrays()

    for sandbox in northrays.list(
        ListSandboxesQuery(
            limit=10,
            labels={"env": "dev"},
            states=[SandboxState.STARTED],
            sort=SandboxListSortField.CREATEDAT,
            order=SandboxListSortDirection.DESC,
        )
    ):
        print(sandbox.id)


if __name__ == "__main__":
    main()
